azkaban-common/src/test/java/azkaban/executor/ExecutionFlowDaoTest.java

```diff
@@ -762,6 +762,33 @@ public void testFlowStatusWithFetchRecentlyFinishedFlows() throws Exception {
     assertThat(finishedFlows2.get(0).getStatus()).isEqualTo(Status.FAILED);
   }
 
+  /**
+   * Test the resiliency of ExecutableFlow when Sla Option is set to NULL.
+   * Make sure that the system overcomes it and the flow proceeds to next valid state.
+   */
+  @Test
+  public void testUpdateExecutableFlowNullSLAOptions() throws Exception {
+    final ExecutableFlow flow = createTestFlow();
+    this.executionFlowDao.uploadExecutableFlow(flow);
+
+    final ExecutableFlow fetchFlow =
+        this.executionFlowDao.fetchExecutableFlow(flow.getExecutionId());
+
+    // set null sla option
+    fetchFlow.getExecutionOptions().setSlaOptions(null);
+    // Try updating flow
+    try {
+      this.executionFlowDao.updateExecutableFlow(fetchFlow);
+    } catch (ExecutorManagerException e) {
+       assert e.getMessage().contains("NPE");
+    }
+    // Fetch flow again, the status must be READY not PREPARING
```

Comment:
This comment gives no more information than what is clearly seen in the code itself. 

Out here, rather than saying the status must be ready - it will be helpful to know why the status must be ready.